src/kernel/syscall.c

```diff
@@ -65,13 +80,26 @@ void sys$exit(i32 return_code)
   asm volatile("jmp process_kernel_return");
 }
 
-void sys$write(u32 fd, char *buffer, size_t length)
+int sys$write(u32 fd, char *buffer, size_t length)
 {
   if (fd == 1)
   {
     for (size_t i = 0; i < length; i++)
     {
       terminal_putchar(buffer[i]);
     }
+    return length;
+  }
+  return -1; // Fail if fd != 1
```

Comment:
I think we should just `ASSERT_ALWAYS()` here because the situation will require some action from me and needs to be recognized.